base_layer/core/src/chain_storage/blockchain_database.rs

```diff
@@ -1150,6 +1183,73 @@ fn find_strongest_orphan_tip<T: BlockchainBackend>(
     Ok((best_accum_difficulty, best_tip_hash))
 }
 
+// Discards the orphan block with the minimum height from the block orphan pool to maintain the configured orphan pool
+// storage limit.
+fn cleanup_orphans_single<T: BlockchainBackend>(
+    db: &mut RwLockWriteGuard<T>,
+    orphan_storage_capacity: usize,
+) -> Result<(), ChainStorageError>
+{
+    if db.get_orphan_count()? > orphan_storage_capacity {
+        trace!(
+            target: LOG_TARGET,
+            "Orphan block storage limit reached, performing simple cleanup.",
+        );
+        let mut min_height: u64 = u64::max_value();
+        let mut remove_hash: Option<BlockHash> = None;
+        db.for_each_orphan(|pair| {
+            let (_, block) = pair.unwrap();
+            if block.header.height < min_height {
+                min_height = block.header.height;
+                remove_hash = Some(block.hash());
```

Comment:
If we have this to save time, we not put in a break here?
We dont have to go through the entire db then .

Reply:
You won't be able to use a break as you need to go through the entire set to find the orphan block with the minimum height. The remove_hash will be replaced when a better alternative is found.